### PR TITLE
feat(qiskit-extension): qiskit simulation extension

### PR DESCRIPTION
**Pull request by str-anger**
> The Qiskit plugin should provide one action: Simulate Circuit.
We have 2! Simulation AND plotting

> The user SHOULD be able to select the following parameters:
> The circuit SHOULD be provided as a QASM file (optionally plugin MAY support .qpy files [qpy | IBM Quantum Documentation](https://docs.quantum.ibm.com/api/qiskit/qpy) )

Qasm2 and Qasm3 supported

> Simulator type (dropdown)

Yes

> Shot count (int)

yes

> 'Memory' (boolean) [this refers to whether the results of all shots individually are saved, or whether they are averaged and batched]

yes

> Noise model (dropdown) [select from several pre-built IBM fake noise models that simulate performance of IBM's real machines e.g. IBM_Vigo]

as agreed, we use noisy simulators, and we documented where you may read about them

> These parameters are all passed directly to the Qiskit calls - they are described in the tutorials but I am happy to clarify any behaviour that isn't clear or could be expanded.

not sure if this is a requirements, but yes they are =)

> The results SHOULD be passed to the user through stdout (either summarised or full depending on the "memory" setting). The results should also be persisted to a file (such as JSON etc) for future analysis if needed.

STDOUT contains statistics (dict with shot distribution), measurements go to a file

> We SHOULD also save one of the images to the experiment - filename should be timestamped in case of multiple executions of the plugin
> for example, maybe a standard histogram of the results

no timestamp, name is provided by a user

### CODE REVIEW

**Review by gabrielg42 — APPROVED** [review]

File: .github/workflows/pull_request.yaml [hunk 1]
```diff
@@ -25,4 +25,7 @@ jobs:
 
       - name: Run extensions smoke tests
         run: >
-          sh ./tests/mock.sh
```
Comment (safeamiiir) [inline .github/workflows/pull_request.yaml:28]: Why are we using this instead?

File: py-qiskit-simulation/files.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    path = Path(directory) / filename
+    print(f"File downloaded successfully: {path}")
+    return path
+
+
+def upload_to_aqueduct(
+        api: API,
+        experiment_id: str,
+        file: str | Path):
+    """Upload a file to an Aqueduct.
```
Comment (safeamiiir) [inline py-qiskit-simulation/files.py]:
```suggestion
    """Upload a file to Aqueduct.
```

**Review by safeamiiir — COMMENTED** [review]
I was not able to run the extension with no change. what I've done:
- checked out into the same branch
- ran the extension
- `.aqueduct-extension-venv` has been created but
- Logfile:
```
- stderr:
Traceback (most recent call last):
  File "/workspace/external/extensions/py-qiskit-simulation/qiskit_simulator.py", line 5, in <module>
    import qiskit.providers.fake_provider
ModuleNotFoundError: No module named 'qiskit'
```

File: .github/workflows/pull_request.yaml (hunk 1 above)
Comment (str-anger) [reply .github/workflows/pull_request.yaml:28]: To run proper automatic tests we need at least one reliable testing installation of aqueduct. Unfortunately, it was not working for some time.

**Review by jatinriverlane — APPROVED** [review]
Thanks Stas

**Pull request merged**